Sources/IRGen/ValueIRGen.swift

```diff
@@ -174,6 +172,40 @@ extension IRGenerator {
       return visitFuncCallExpr(expr)
     }
   }
+
+  @discardableResult
+  func codegenImplicitCopy(_ value: IRValue, expr: Expr, destType: DataType) -> IRValue {
+    let canTy = context.canonicalType(expr.type!)
+    let canDestTy = context.canonicalType(destType)
+
+    if case .any = canDestTy {
+      return codegenPromoteToAny(value: value, type: canTy)
+    }
+
+    if context.isTriviallyCopyable(canTy) {
+      return value
+    }
+
+    guard let decl = context.decl(for: canTy) else {
+      fatalError("no decl for non-trivially-copyable type?")
+    }
+
+    if decl.isIndirect {
+      let irType = resolveLLVMType(canDestTy)
+      let allocaForRelease = createEntryBlockAlloca(currentFunction!.functionRef!,
+                                                    type: irType,
+                                                    name: "releasable",
+                                                    storage: .reference)
+      allocaForRelease.write(value)
+      codegenRetain(value)
+      releaseCalls.append({ [unowned self] in
```

Comment:
we need to only be adding a release call if its a var assign exr of local scope, I think?